src/NHibernate/Connection/ResilientDriverConnectionProvider.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using NHibernate;
+using NHibernate.Connection;
+
+namespace NHibernate.Connection
+{
+	public class ResilientDriverConnectionProvider : DriverConnectionProvider
+	{
+		public const String ConnectionDelayBetweenTries = "connection.delay_between_tries";
+		public const String ConnectionMaxTries = "connection.max_tries";
+
+		private static readonly IInternalLogger log = LoggerProvider.LoggerFor(typeof(ResilientDriverConnectionProvider));
+
+		public ResilientDriverConnectionProvider()
+		{
+			this.MaxTries = 3;
+			this.DelayBetweenTries = TimeSpan.FromSeconds(5);
+		}
+
+		public Int32 MaxTries { get; set; }
+
+		public TimeSpan DelayBetweenTries { get; set; }
+
+		public override void Configure(IDictionary<String, String> settings)
+		{
+			String maxTries;
+			String delayBetweenTries;
+
+			if (settings.TryGetValue(ConnectionMaxTries, out maxTries) == true)
```

Comment:
Is it a NHibernate accepted practice to "explicit" boolean conditions like this and the three next conditions?  
(I personally quite dislike it. Even on nullable boolean I would rather do `if (thingYieldingNullableBoolean ?? false)` instead of comparing to true. By the way, I dislike non mandatory usages of `this.` too. Even more than now VS signal them as useless.)